general-concepts/use-flags/text.xml

```diff
@@ -5,113 +5,179 @@
-	dependency is not detected by the package manager tools and can easily
-	break, among other issues.
+Packages should not configure and link based upon what is available at
+compile time <d/> any autodetection must be overridden. This is commonly
+referred to as the dependency being "automagic" - This is bad because the
+dependency is not detected by the package manager tools and can easily
+break, among other issues.
 </p>
 
+<note>
+The status of USE flags is saved in the VDB, and their value in
+<c>pkg_prerm</c> and <c>pkg_postrm</c> is taken from there. This means that
+setting or unsetting a USE flag between merge and unmerge has no effect.
+</note>
+</body>
+
+<section>
+<title>When not to use USE flags?</title>
+<body>
 <p>
-	The usage of a <c>USE</c> flag should not control runtime dependencies when
-	the package does not link to it. Doing so will create extra
-	configuration for the package and re-compilation for no underlying file
-	change on disk. This should be avoided and instead can be conveyed to the
-	user via post install messages if needed.
+While <c>USE</c> flags are generally considered beneficial to users, there
+are valid use cases for avoiding them. When writing ebuilds, you should
+consider whether you should add flags for particular conditional features,
+or explore one of the alternative solutions.
 </p>
 
-<note>
-	The status of USE flags is saved in the VDB, and their value in
-	<c>pkg_prerm</c> and <c>pkg_postrm</c> is taken from there. This means that
-	setting or unsetting a USE flag between merge and unmerge has no effect.
-</note>
+<p>
+The usage of a <c>USE</c> flag should not control runtime dependencies when
+the package does not link to it. Doing so will create extra
+configuration for the package and re-compilation for no underlying file
+change on disk. This should be avoided and instead can be conveyed to the
+user via post install messages if needed.
+</p>
+
+<p>
+<c>USE</c> flags must not be used to control installing files that are small,
+non-intrusive, do not introduce additional build-time dependencies or cause
+significant increase of build time. Examples of such files are bash completion
+files, init.d scripts, systemd service files. The rationale is same as above.
+Instead, those files must be installed unconditionally.
+</p>
+
+<p>
+A similar case can be made for packages having multiple conditional programs
+or modules. Whenever they could result in a large number of <c>USE</c> flags
+that would force the user to spend a lot of time choosing the necessary flags
+and possibly rebuilding after incomplete choices, you should consider
+reducing the use of flags to those programs or modules that have external
+dependencies and/or long build times. The rest of them should be built
+unconditionally instead, or controlled by a flag such as <c>minimal</c>.
+</p>
+
+<p>
+It is customary not to introduce USE flags that merely manipulate
```

Comment:
I had a hard time reading "customary not to....". I think the phrase "it is customary to" is rather common in english, but "customary not to" is comparatively rare. So when the brain sees that combination of letters, it reads "customary to" regardless of what the text actually says. In any case, this wording could be stronger: "you should not..." and that solves my problem too.

Reply:
Done.